src/Component/Metadata/Resource.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Metadata;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class Resource
+{
+    private ?Operations $operations;
+
+    public function __construct(
+        private ?string $alias = null,
+        ?array $operations = null,
+    ) {
+        $this->operations = null === $operations ? null : new Operations($operations);
```

Comment:
will it be that hard just to expect the Operations class to be passed here? What is blocking us from it? Nested attributes?

Reply:
That's will be less user friendly for DX.